Mathlib/NumberTheory/NumberField/Cyclotomic/Basic.lean

```diff
@@ -692,6 +691,93 @@ theorem absdiscr_prime [IsCyclotomicExtension {p} ℚ K] :
   rw [absdiscr_prime_pow_succ p 0 K]
   simp [Nat.sub_sub]
 
+theorem natAbs_absdiscr_aux₁ (hk : 0 < k) :
```

Comment:
It seems unlikely that anyone else would use this, but if it's easy you can move it to a more basic file.

Reply:
Also, you can maybe suffer a little bit less using `k` instead of `k-1` and `k+1` instead of `k`.

Reply:
Something like
```lean
open Nat Finset in
theorem natAbs_absdiscr_aux₁ :
    p ^ (p ^ k * ((p - 1) * (k + 1) - 1)) =
      (p ^ (k + 1) : ℕ) ^ (p ^ (k + 1)).totient /
        ∏ q ∈ (p ^ (k + 1)).primeFactors, q ^ ((p ^ (k + 1)).totient / (q - 1)) := by
  have := hp.out.two_le
  have H : 1 ≤ (k + 1) * (p - 1) := Right.one_le_mul (by cutsat) <| le_sub_one_of_lt <| hp.1.one_lt
  have h :  p ^ k ≤ (k + 1) * (p ^ k * (p - 1)) :=
    mul_left_comm (k + 1) _ _ ▸ le_mul_of_one_le_right (Nat.zero_le _) H
  rw [totient_prime_pow hp.1 (by cutsat), primeFactors_prime_pow (by cutsat) hp.1, prod_singleton,
    mul_div_left _ (by cutsat), ← pow_mul, add_tsub_cancel_right]
  qify [pow_dvd_pow (p : ℤ) h]
  field_simp
  rw [← pow_add]
  congr
  zify [mul_comm (p - 1) _ ▸ H]
  ring
```

Reply:
> Also, you can maybe suffer a little bit less using `k` instead of `k-1` and `k+1` instead of `k`.

Yes, that's a good point. Thanks for the improved proof!